src/equations/compressible_euler_multicomponent_2d.jl

```diff
@@ -7,61 +7,50 @@
-  gamma1          ::RealT 
-  gamma2          ::RealT
-  gas_constant1   ::RealT
-  gas_constant2   ::RealT
-  cv1             ::RealT
-  cv2             ::RealT
-  cp1             ::RealT
-  cp2             ::RealT
+
+struct CompressibleEulerMulticomponentEquations2D{NCOMP, NVARS, RealT<:Real} <: AbstractCompressibleEulerMulticomponentEquations{2, NVARS}
+  gamma             ::SVector{NCOMP, RealT} 
+  gas_constant      ::SVector{NCOMP, RealT}
+  cv                ::SVector{NCOMP, RealT}
+  cp                ::SVector{NCOMP, RealT}
+  CompressibleEulerMulticomponentEquations2D{NCOMP, NVARS, RealT}(gamma, gas_constant) where {NCOMP, NVARS, RealT<:Real} = new(gamma, gas_constant, gas_constant./(gamma.-1), gas_constant .+ gas_constant./(gamma .-1))
 end
 
-function CompressibleEulerMulticomponentEquations2D()
 
-  # Set ratio of specific heat of the gas per species 
-  # (It holds: gamma = cp/cv) (Can be used for consistency check!)
-  gamma1  = 1.4
-  gamma2  = 1.4
-  
-  # Set specific gas constant with respect to the molar mass per species 
-  # (It holds: gas_constant = R/M, with R = molar gas constant, M = molar mass) 
-  # (For a calorically and thermally perfect gas Mayer's relation holds: gas_constant = cp - cv) (Can be used for consistency check!)
-  gas_constant1     = 0.4
-  gas_constant2     = 0.4
+function CompressibleEulerMulticomponentEquations2D(; gamma, gas_constant)
 
-  # Set specific heat for a constant volume per species
-  cv1    = 1.0
-  cv2    = 1.0
+  _gamma        = promote(gamma...)
+  _gas_constant = promote(gas_constant...)
+  T             = promote_type(eltype(_gamma), eltype(_gas_constant))
 
-  # Set specific heat for a constant pressure per species
-  cp1    = 1.4
-  cp2    = 1.4
+  length(_gamma) == length(_gas_constant) || throw(DimensionMismatch("gamma and gas_constant should have the same length"))
+  NVARS = length(_gamma) + 3
+  NCOMP = length(_gamma)
 
-  CompressibleEulerMulticomponentEquations2D(gamma1, gamma2, gas_constant1, gas_constant2, cv1, cv2, cp1, cp2)
+  return CompressibleEulerMulticomponentEquations2D{NCOMP, NVARS, T}(map(T, _gamma), map(T, _gas_constant))
 end
 
 
-function CompressibleEulerMulticomponentEquations2D(gamma1, gamma2, gas_constant1, gas_constant2)
-  
-  # Set specific heat for a constant volume per species
-  cv1    = gas_constant1 / (gamma1 - 1.0)
-  cv2    = gas_constant2 / (gamma2 - 1.0)
+get_name(::CompressibleEulerMulticomponentEquations2D) = "CompressibleEulerMulticomponentEquations2D"
 
-  # Set specific heat for a constant pressure per species
-  cp1    = gas_constant1 + cv1 
-  cp2    = gas_constant2 + cv2
 
-  CompressibleEulerMulticomponentEquations2D(gamma1, gamma2, gas_constant1, gas_constant2, cv1, cv2, cp1, cp2)
+function varnames(::typeof(cons2cons), equations::CompressibleEulerMulticomponentEquations2D{NCOMP, NVARS}) where {NCOMP, NVARS}
+  
+  cons  = ("rho_v1", "rho_v2", "rho_e")
+  rhos  = ntuple(n -> "rho" * string(n), Val(NCOMP))
```

Comment:
done in 657c2b2

Reply:
Yeah, this sounds good to me

Reply:
This can't work since we got `AbstractCompressibleEulerMulticomponentEquations{NDIMS, NVARS}` instead of `AbstractCompressibleEulerMulticomponentEquations{NCOMP, NVARS}` or am I wrong?

Reply:
Right. What about changing `AbstractC...` to
```julia
abstract type AbstractCompressibleEulerMulticomponentEquations{NDIMS, NVARS, NCOMP} <: AbstractEquations{NDIMS, NVARS} end
```
instead, and updating `ncomponents(...)` accordingly?
```

Reply:
done in c7e9a75